From 00d593a93cfcfebd6a0b8884d13d134c3b3a8f6b Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Tue, 10 May 2016 05:05:11 +0800
Subject: [PATCH 121/143] [VPG]: drm/i915: Retry when force wake time out

Some platforms have force wake time out issue, retry several times
to avoid it.
---
 drivers/gpu/drm/i915/intel_uncore.c |   13 ++++++++++---
 1 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_uncore.c b/drivers/gpu/drm/i915/intel_uncore.c
index d62bfbd..c8d5a62 100644
--- a/drivers/gpu/drm/i915/intel_uncore.c
+++ b/drivers/gpu/drm/i915/intel_uncore.c
@@ -101,11 +101,18 @@ fw_domain_get(const struct intel_uncore_forcewake_domain *d)
 static inline void
 fw_domain_wait_ack(const struct intel_uncore_forcewake_domain *d)
 {
+	int count = 0;
+retry:
 	if (wait_for_atomic((__raw_i915_read32(d->i915, d->reg_ack) &
 			     FORCEWAKE_KERNEL),
-			    FORCEWAKE_ACK_TIMEOUT_MS))
-		DRM_ERROR("%s: timed out waiting for forcewake ack request.\n",
-			  intel_uncore_forcewake_domain_to_str(d->id));
+			     FORCEWAKE_ACK_TIMEOUT_MS)) {
+		if (++count >= 5) {
+			DRM_ERROR("%s: timed out waiting for forcewake ack request.\n",
+				  intel_uncore_forcewake_domain_to_str(d->id));
+		} else {
+			goto retry;
+		}
+	}
 }
 
 static inline void
-- 
1.7.1

